### PR TITLE
Partly fixes #158 Display information for continuous deployment

### LINKED ISSUE #158: Register repositories from WebUI for continuous deployment
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

The documentation is deployed to gh-pages, but after that deployment, there is no redeployment at each commit/change in the docs.
### Expected behaviour

Docs should be redeployed after each change in them

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

Refering to https://github.com/fossasia/yaydoc/issues/158#issuecomment-309700871, we are showing  information about continuous deployment of documentation.
## Related Issue

Partly fixes #158 
## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

File: views/index.jade [hunk 1]
```diff
@@ -37,14 +37,32 @@ block content
...
+        .modal-header
+          button.close(type='button', data-dismiss='modal') ×
+          h5.modal-title Github Deployment
+        .modal-body
+          p
+            | In order to enable continuous deployment of Documentation, add the following content to your &nbsp;
+            code .travis.yml
+          pre
+            | deploy
+            | - source &lt;(curl -s https://raw.githubusercontent.com/fossasia/yaydoc/master/generate_ci.sh)
```
Comment (pri22296) [inline views/index.jade]: This gives the impression to a user that this is enough to setup continuous deployment when infact they also need to create a environment variable OAUTH_TOKEN or setup ssh. add a point mentioning that. Provide enough information here that enables deployment successfully without looking at the docs.

File: views/index.jade [hunk 2]
```diff
@@ -37,14 +37,47 @@ block content
...
+            | , making necessary changes
+          pre
+            | # Add environment variable following description in http://yaydoc.fossasia.org/#usage
+            | env:
+            |   global:
+            |     - AUTHOR=&lt;author-name&rt;
+            |     - DOCPATH=&lt;doc-path&rt;
+            |     - PROJECTNAME=&lt;project-name&rt;
+            |     - VERSION=&lt;version-number&rt;
+            |     - ENCRYPTION_KEY=&lt;32 character alphanumeric string&rt;
```
Comment (pri22296) [inline views/index.jade]: why is this needed? we only need it in our repo to run tests right?

File: views/index.jade [hunk 3]
```diff
@@ -37,14 +37,47 @@ block content
...
+            code .travis.yml
+            | , making necessary changes
+          pre
+            | # Add environment variable following description in http://yaydoc.fossasia.org/#usage
+            | env:
+            |   global:
+            |     - AUTHOR=&lt;author-name&rt;
+            |     - DOCPATH=&lt;doc-path&rt;
+            |     - PROJECTNAME=&lt;project-name&rt;
+            |     - VERSION=&lt;version-number&rt;
```
Comment (pri22296) [inline views/index.jade]: Maybe include here only required variables not optional?

File: views/index.jade (hunk 2 above)
Comment (imujjwal96) [inline views/index.jade]: it is needed in the crypt function. A user need to set this variable to enable github deployment, as this is used in encrypting the personal access token

Comment (pri22296) [inline views/index.jade]: That's only needed from the webUI right? not by the travis implementation!

Comment (imujjwal96) [inline views/index.jade]: Oh.OK!!!! :+1:

File: views/index.jade (hunk 3 above)
Comment (imujjwal96) [inline views/index.jade]: It seems that the only required variable DOCPATH.

Comment (pri22296) [inline views/index.jade]: well even DOCPATH isn't required. It defaults to 'docs'

Comment (imujjwal96) [inline views/index.jade]: then?

**Comment by pri22296** [comment]
@imujjwal96 Considering https://github.com/fossasia/yaydoc/issues/158#issuecomment-309707865 , It seems that the travis cli takes a parameter -t to login. would it be possible to encrpyt the token we receive and add it to the message displayed?

File: views/index.jade (hunk 3 above)
Comment (pri22296) [inline views/index.jade]: Good point. let's keep DOCPATH. should we encourage the user to use `.yaydoc.yml` or `.travis.yml`

Comment (imujjwal96) [inline views/index.jade]: Later in development, we can automate this process and add the `.yaydoc.yml` file in the user's repository.

Comment (pri22296) [inline views/index.jade]: cool

**Comment by imujjwal96** [comment]
> encrpyt the token we receive

Which token are you referring to?

**Comment by pri22296** [comment]
The github token we recieve when deploy is clicked

**Comment by imujjwal96** [comment]
image

We can use this approach. It doesn't mention the requirement to login

**Comment by imujjwal96** [comment]
I've installed travis CLI in Heroku, tried few things, but how couldn't find a way to display the encrypted token to the user. I have, however, been able to add that to the user's .travis.yml file
@pri22296  Any idea on how to proceed from here?

**Comment by pri22296** [comment]
doesn't the travis cli print the encryted var to stdout?

**Comment by imujjwal96** [comment]
I'm now looking for a solution in which we'll be able to let the user edit his or her `.travis.yml` file from our webapp itself.

If we can merge this as it is, I can raise and work on a separate issue

**Comment by pri22296** [comment]
Could you update the screenshot?

**Comment by pri22296** [comment]
`DOCPATH=<doc_path&rt;` fix this

File: views/index.jade [hunk 4]
```diff
@@ -37,14 +37,47 @@ block content
...
+            | # Add SSH Key following description in http://yaydoc.fossasia.org/#using-ssh
+            | before_deploy:
+            | - openssl aes-256-cbc -K $encrypted_37510afbd506_key -iv $encrypted_37510afbd506_iv -in .utility/deploy_key.enc -out .utility/deploy_key -d
+            | - eval "$(ssh-agent -s)"
+            | - chmod 600 .utility/deploy_key
+            | - ssh-add .utility/deploy_key
+            | deploy
+            | - provider: script
+            |   skip_cleanup: true
+            |   script: source &lt;(curl -s https://raw.githubusercontent.com/fossasia/yaydoc/master/generate_ci.sh)
```
Comment (pri22296) [inline views/index.jade]: This won't work for non-python projects. you need to add `pip install --user virtualenv` for non-python projects. Considering that this addition will need to be conditional depending on the language, for now maybe just add a comment regarding that.

Comment (imujjwal96) [inline views/index.jade]: Since deploy can execute only one script, i guess we'll have to mention that the user must add this in the before_deploy block

Comment (pri22296) [inline views/index.jade]: #182 should fix this problem. views?

Comment (imujjwal96) [inline views/index.jade]: isn't 182 the ~ISSUE~ ? jk. 
Totally agree with you. You just resolved the concern that I had, as there are many webapp based on other languages with CSS/JS as their primary language according to Github

Comment (imujjwal96) [inline views/index.jade]: Solving that issue wouldn't require the change you request, right?

Comment (pri22296) [inline views/index.jade]: yep. ignore this then

Comment (pri22296) [inline views/index.jade]: I tried to test this line in my test repo. I'm getting the following error
sh: 1: Syntax error: "(" unexpected
build log: https://travis-ci.org/pri22296/testYaydoc2/builds/246565861
travis.yml: https://github.com/pri22296/testYaydoc2/blob/master/.travis.yml

Comment (imujjwal96) [inline views/index.jade]: Why is this executing `sh`?

File: views/index.jade [hunk 5]
```diff
@@ -37,14 +37,47 @@ block content
...
+            | env:
+            |   global:
+            |     - DOCPATH=&lt;doc-path&gt;
+            | # Add SSH Key following description in http://yaydoc.fossasia.org/#using-ssh
+            | before_deploy:
+            | - openssl aes-256-cbc -K $encrypted_37510afbd506_key -iv $encrypted_37510afbd506_iv -in .utility/deploy_key.enc -out .utility/deploy_key -d
+            | - eval "$(ssh-agent -s)"
+            | - chmod 600 .utility/deploy_key
+            | - ssh-add .utility/deploy_key
+            | deploy
```
Comment (pri22296) [inline views/index.jade]: add `:` after deploy

File: views/index.jade (hunk 4 above)
Comment (pri22296) [inline views/index.jade]: No idea. since we are sourcing and maybe travis runs the script section under sh? not sure

File: views/index.jade (hunk 5 above)
Comment (imujjwal96) [inline views/index.jade]: Does this fix the error?

Comment (pri22296) [inline views/index.jade]: No. I already changed that in my travis.yml. That error is still not fixed

File: views/index.jade (hunk 4 above)
Comment (imujjwal96) [inline views/index.jade]: let me raise a token in travis' repository.

Comment (pri22296) [inline views/index.jade]: for the time being I have replaced that with
```bash
wget -O - https://raw.githubusercontent.com/fossasia/yaydoc/master/generate_ci.sh | bash
```

Comment (imujjwal96) [inline views/index.jade]: image

File: views/index.jade [hunk 6]
```diff
@@ -37,14 +37,47 @@ block content
...
+            | before_deploy:
+            | - openssl aes-256-cbc -K $encrypted_37510afbd506_key -iv $encrypted_37510afbd506_iv -in .utility/deploy_key.enc -out .utility/deploy_key -d
+            | - eval "$(ssh-agent -s)"
+            | - chmod 600 .utility/deploy_key
+            | - ssh-add .utility/deploy_key
+            | deploy:
+            | - provider: script
+            |   skip_cleanup: true
+            |   script: wget -O - https://raw.githubusercontent.com/fossasia/yaydoc/master/generate_ci.sh | bash
+            |   on:
```
Comment (pri22296) [inline views/index.jade]: maybe remove `on branch` from here?

Comment (imujjwal96) [inline views/index.jade]: why?

Comment (pri22296) [inline views/index.jade]: Because That's not required by Yaydoc. This info belongs in the travis's docs that you can deploy to another branch. If a user wants to to deploy to another branch they would add that line.
